src/auth/credential.ts

```diff
@@ -190,28 +190,36 @@ export interface GoogleOAuthAccessToken {
 function requestAccessToken(client: HttpClient, request: HttpRequestConfig): Promise<GoogleOAuthAccessToken> {
   return client.send(request).then((resp) => {
     const json = resp.data;
-    if (json.error) {
-      let errorMessage = 'Error fetching access token: ' + json.error;
-      if (json.error_description) {
-        errorMessage += ' (' + json.error_description + ')';
-      }
-      throw new FirebaseAppError(AppErrorCodes.INVALID_CREDENTIAL, errorMessage);
-    } else if (!json.access_token || !json.expires_in) {
+    if (!json.access_token || !json.expires_in) {
       throw new FirebaseAppError(
         AppErrorCodes.INVALID_CREDENTIAL,
         `Unexpected response while fetching access token: ${ JSON.stringify(json) }`,
       );
-    } else {
-      return json;
     }
+    return json;
   }).catch((err) => {
-    throw new FirebaseAppError(
-      AppErrorCodes.INVALID_CREDENTIAL,
-      `Failed to parse access token response: ${err.toString()}`,
-    );
+    const message = getErrorMessage(err);
+    throw new FirebaseAppError(AppErrorCodes.INVALID_CREDENTIAL, message);
```

Comment:
You can directly pass `getErrorMessage(err)` instead of `message` interim variable.

Reply:
Done